GPT/beta-commands/gpt-function-calling.py

```diff
@@ -122,38 +121,44 @@ def gpt_function_query(prompt: str, content: str, insert_response: Callable[[str
 
     else:
         notify("GPT Failure: Check API Key, Model, or Prompt")
-        print(response.json())
+        raise Exception(f"GPT Failure at POST request: {response.json()}")
 
 def process_function_calls(insert_response, message):
     try:
         tool_calls = message["tool_calls"]
-        while tool_calls:
-            tool = tool_calls.pop()
-            first_argument = tool['function']['arguments']
-            try:
-                first_argument = json.loads(first_argument)['str']
-            except Exception as e:
-                notify(f"Argument Jason was malformed: {e}")
-            match tool['function']['name']:
-                case 'insert':
-                    insert_response(first_argument)
-                case 'display':
-                    display_response(first_argument)
-                case 'notify':
-                    notify_user(first_argument)
-                case 'search_for_command':
-                    search_for_command(first_argument)
     except Exception as e:
-        notify(f"No tool_calls found in response from LLM: {e}")
+        notify(f"No tool calls were found in LLM response")
+        return
+
+    for tool in tool_calls:
+
+        try: 
+            first_argument = tool['function']['arguments']
+            first_argument = json.loads(first_argument)['str']
+        except Exception as e:
+            notify(f"Argument JSON was malformed: {e}")
+            # Try next tool call if this one fails
+            continue
```

Comment:
Is this desirable? If the first call fails, I'm not sure we want to blindly go on to the next one. Presumably the model put them in this order for a reason?